client/packages/host/src/components/Login/Login.tsx

```diff
@@ -55,22 +55,44 @@ export const Login = () => {
     }
   }, [error]);
 
-  const loginErrorMessage = useMemo(() => {
-    if (!error) return '';
+  const loginError: { error: string; hint?: string } = useMemo(() => {
+    if (!error) return { error: '' };
+
+    if (error.message === 'ConnectionError') {
+      return {
+        error: t('error.connection-error'),
+        hint: t('error.connection-error-hint'),
+      };
+    }
+
     if (error.message === 'AccountBlocked') {
-      if (timeoutRemaining < 1000) return '';
+      if (timeoutRemaining < 1000) return { error: '' };
 
       const formattedTime = customDate(
         new Date(0, 0, 0, 0, 0, 0, timeoutRemaining),
         'm:ss'
       );
-      return `${t('error.account-blocked')} ${formattedTime}`;
+      return { error: `${t('error.account-blocked')} ${formattedTime}` };
+    }
+
+    if (error.message === 'InvalidCredentials') {
```

Comment:
I was trying to suggest that we can make this logic a bit more typed by not saving __typename to string [here](https://github.com/msupply-foundation/open-msupply/blob/4b448ffce5965e1ef808fb98b4f329a80794a046/client/packages/common/src/authentication/api/api.ts#L33-L44), but then got lost in tracing through context, helpers, and useEffects : (

Reply:
Also its' a bit wiered how we add timeoutRemaining to the AuthenticationError shape, with undefined, when we know it's present with __typename === 'AccountBlocked'

Reply:
Yeah I don't love the strings, and I found the different layers of mapping confusing... didn't want to touch too much and bloat this PR but would be nice to come back to in another refactor issue?

Reply:
Sure, i have some thoughts about doing cookie setting on back end